### PR TITLE
RMB-602: PostgresClient.update futurisation

### PR DESCRIPTION
**Pull request by julianladisch**
This also extracts duplicate code into
withConn(AsyncResult<SQLConnection>, Function<Conn, Future<T>>).

### CODE REVIEW

File: domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java [hunk 1]
```diff
@@ -434,4 +438,175 @@ static String durationMsg(String descriptionKey, String sql, long startNanoTime)
...
+    }
+    return update(table, entity, PostgresClient.DEFAULT_JSONB_FIELD_NAME, where, returnUpdatedIds);
+  }
+
+  /**
+   * Update all records in {@code table} that match the {@code CQLWrapper} query.
+   * @entity new content for the matched records
+   * @return one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public Future<RowSet<Row>> update(String table, Object entity, CQLWrapper filter, boolean returnUpdatedIds) {
```
Comment (hjiebsco) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java:550]: Fix Javadoc

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java [hunk 2]
```diff
@@ -1219,31 +1192,29 @@ private void saveBatch(AsyncResult<SQLConnection> sqlConnection, boolean upsert,
...
 
   /**
-   * update a specific record associated with the key passed in the id arg
+   * Update a specific record associated with the key passed in the id arg
+   * @param table - table to save to (must exist)
+   * @param entity - pojo to save
+   * @param id - key of the entity being updated
+   */
+  public Future<RowSet<Row>> update(String table, Object entity, String id) {
+    return withConn(conn -> conn.update(table, entity, id));
```
Comment (hjiebsco) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1206]: Add `return` to Javadoc

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java [hunk 3]
```diff
@@ -1295,76 +1266,141 @@ public void update(String table, Object entity, String id, Handler<AsyncResult<R
...
+                     Handler<AsyncResult<RowSet<Row>>> replyHandler) {
+    update(table, entity, filter, returnUpdatedIds).onComplete(replyHandler);
+  }
+
+  /**
+   * Update all records in {@code table} that match the {@code CQLWrapper} query.
+   * @param entity new content for the matched records
+   * @return one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public Future<RowSet<Row>> update(String table, Object entity, CQLWrapper filter, boolean returnUpdatedIds) {
```
Comment (hjiebsco) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1336]: Fix Javadoc

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java [hunk 4]
```diff
@@ -1295,76 +1266,141 @@ public void update(String table, Object entity, String id, Handler<AsyncResult<R
...
+    return withConn(conn -> conn.update(table, entity, filter, returnUpdatedIds));
   }
 
+  /**
+   * Update all records in {@code table} that match the {@code CQLWrapper} query.
+   * @param entity new content for the matched records
+   * @param replyHandler one row with the id for each updated record if returnUpdatedIds is true
+   */
   public void update(String table, Object entity, CQLWrapper filter, boolean returnUpdatedIds,
```
Comment (hjiebsco) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1345]: Fix Javadoc

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java [hunk 5]
```diff
@@ -1295,76 +1266,141 @@ public void update(String table, Object entity, String id, Handler<AsyncResult<R
...
+    update(table, entity, filter, returnUpdatedIds).onComplete(replyHandler);
   }
 
-  public void update(AsyncResult<SQLConnection> conn, String table, Object entity, CQLWrapper filter,
+  /**
+   * Update all records in {@code table} that match the {@code CQLWrapper} query.
+   * @param entity new content for the matched records
+   * @param replyHandler one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public void update(AsyncResult<SQLConnection> sqlConnection, String table, Object entity, CQLWrapper filter,
```
Comment (hjiebsco) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1355]: Fix Javadoc

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java [hunk 6]
```diff
@@ -1295,76 +1266,141 @@ public void update(String table, Object entity, String id, Handler<AsyncResult<R
...
+  /**
+   * Update all records in {@code table} that match the SQL {@code whereClause}.
+   *
+   * <p>Danger: The {@code whereClause} is prone to SQL injection. Consider using
+   * an {@code update} method that takes {@link CQLWrapper} or {@link Criterion}.
+   *
+   * @param entity new content for the jsonbField of the matched records
+   * @param replyHandler one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public void update(AsyncResult<SQLConnection> sqlConnection, String table, Object entity, String jsonbField,
```
Comment (hjiebsco) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1371]: Javadoc

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java [hunk 7]
```diff
@@ -1295,76 +1266,141 @@ public void update(String table, Object entity, String id, Handler<AsyncResult<R
...
+  /**
+   * Update all records in {@code table} that match the SQL {@code whereClause} query.
+   *
+   * <p>Danger: The {@code whereClause} is prone to SQL injection. Consider using
+   * an {@code update} method that takes {@link CQLWrapper} or {@link Criterion}.
+   *
+   * @param entity new content for the jsonbField of the matched records
+   * @return one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public Future<RowSet<Row>> update(String table, Object entity, String jsonbField, String whereClause, boolean returnUpdatedIds) {
```
Comment (hjiebsco) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1387]: Javadoc

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java [hunk 8]
```diff
@@ -1295,76 +1266,141 @@ public void update(String table, Object entity, String id, Handler<AsyncResult<R
...
+  /**
+   * Update all records in {@code table} that match the SQL {@code whereClause} query.
+   *
+   * <p>Danger: The {@code whereClause} is prone to SQL injection. Consider using
+   * an {@code update} method that takes {@link CQLWrapper} or {@link Criterion}.
+   *
+   * @param entity new content for the jsonbField of the matched records
+   * @param replyHandler one row with the id for each updated record if returnUpdatedIds is true
+   */
   public void update(String table, Object entity, String jsonbField, String whereClause, boolean returnUpdatedIds,
```
Comment (hjiebsco) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1400]: Javadoc

**Review by hjiebsco — APPROVED** [review]
Some Javadoc can be more complete.

File: domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java (hunk 1 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java]: `@param` javadoc is not needed if the param is sufficiently explained in the summary: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java (hunk 2 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java]: Added.

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java (hunk 3 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1337]: `@param` javadoc is not needed if the param is sufficiently explained in the summary: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java (hunk 4 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1346]: `@param` javadoc is not needed if the param is sufficiently explained in the summary: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java (hunk 5 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1356]: `@param` javadoc is not needed if the param is sufficiently explained in the summary: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java (hunk 6 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1372]: `@param` javadoc is not needed if the param is sufficiently explained in the summary: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java (hunk 7 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1388]: `@param` javadoc is not needed if the param is sufficiently explained in the summary: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java (hunk 8 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java:1401]: `@param` javadoc is not needed if the param is sufficiently explained in the summary: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

**Pull request merged**